src/components/limited-restoration-panel/LimitedRestorationPanel.vue

```diff
@@ -1,127 +1,148 @@
-          class="number-months-field"
+          id="text-field-months"
+          v-model="inputValue"
+          class="text-field-months"
           type="number"
           dense
+          filled
           hide-spin-buttons
-          :rules="monthRules"
-          v-model="numberOfMonths"
-          :disabled="!customMonths"
-          @change="onMonthsChanged"
-          filled />
+          :rules="monthsRules"
+          :disabled="(radioValue !== 'customMonths')"
+          @input="onMonthsInput($event)"
+        />
       </v-form>
-      <div class="ml-2 mt-2 month-text">month(s)</div>
+      <span class="ml-2 mt-2 month-text">month(s)</span>
     </v-row>
   </v-radio-group>
 </template>
 
 <script lang="ts">
-import { Component, Emit, Mixins, Prop, Watch } from 'vue-property-decorator'
-import { DateMixin } from '@/mixins' // NB: local mixin (StoryBook can't find it otherwise)
+import Vue from 'vue'
+import { Component, Emit, Prop, Watch } from 'vue-property-decorator'
 import { FormIF } from '@bcrs-shared-components/interfaces'
 
 @Component({})
-export default class LimitedRestorationPanel extends Mixins(DateMixin) {
+export default class LimitedRestorationPanel extends Vue {
   // Refs
   $refs!: Vue['$refs'] & {
     monthsRef: FormIF
   }
 
-  @Prop({ default: '' }) readonly currentDate!: string
-
-  @Prop({ default: '' }) readonly expiryDate!: string
-
+  @Prop({ required: true }) readonly months!: number
   @Prop({ default: 24 }) readonly maxNumberOfMonths!: number
 
   // Local properties
-  private customMonths = ''
-  private selectMonths = ''
-  private numberOfMonths: number = null
-  private monthRules: ((data: string) => void)[] = []
+  radioValue = '' // v-model for radio buttons
+  inputValue = '' // v-model for months text field
+
+  /**
+   * The validation rules for months input.
+   */
+  get monthsRules (): Array<(v: string) => boolean | string> {
+    return [
+      (v) => (+v > 0 && +v <= this.maxNumberOfMonths) || 'Must be between 1 and ' + this.maxNumberOfMonths,
+      (v) => (+v % 1 === 0) || 'Must be a whole number'
+    ]
+  }
 
   /**
    * Called when the component is mounted.
-   * Auto select the previous months input of user (if any).
-   * If there is no previous input, auto select 24 months.
+   * Sets the initial radio button and months input (if applicable).
    */
   mounted (): void {
-    this.setMonthRules()
-    const draftMonths = this.subtractDates(this.currentDate, this.expiryDate)
-    if (draftMonths !== 24 && draftMonths !== 18 && draftMonths !== 12 && draftMonths !== 6) {
-      this.selectMonths = 'customMonths'
-      this.numberOfMonths = draftMonths
-      this.setCustomMonths(this.numberOfMonths)
-      // Emit validation on load
-      if (this.numberOfMonths > this.maxNumberOfMonths || this.numberOfMonths < 1) {
-        this.monthsValid(false)
-      }
+    if ([24, 18, 12, 6].includes(this.months)) {
+      this.radioValue = this.months.toString()
```

Comment:
I really really like this. We would have to keep in mind to change the panel in Create and Edit accordingly since the implementation changed. We're going to have to create tickets for those right?

Reply:
I'm updating this component specifically for Edit UI (part of my ticket).

Yes, if Create UI has build errors in this component then it will need to import the latest component (this one). In the meantime, it can keep importing the previous version of this component.

Reply:
Alright, noted.

Reply:
> Never mind, I just saw that you've made the change for Edit UI in the cleanup PR! 😃

I have it working but not quite finished!